integration/network/bridge/iptablesdoc/templates/usernet-portmap-routed.md

```diff
@@ -0,0 +1,53 @@
+## Container on a routed-mode network, with a published port
+
+Running the daemon with the userland proxy disabled then, as before, adding a network running a container with a mapped port, equivalent to:
+
+	docker network create \
+	  -o com.docker.network.bridge.name=bridge1 \
+	  -o com.docker.network.bridge.gateway_mode_ipv4=routed \
+	  --subnet 192.0.2.0/24 --gateway 192.0.2.1 bridge1
+	docker run --network bridge1 -p 8080:80 --name c1 busybox
+
+The filter table is the same as with the userland proxy enabled.
+
+_Note that this means inter-network communication is disabled as-normal so,
+although published ports will be directly accessible from a remote host
+they are not accessible from containers in neighbouring docker networks
+on the same host._
+
+<details>
+<summary>Filter table</summary>
+
+    {{index . "LFilter4"}}
+
+    {{index . "SFilter4"}}
+
+</details>
+
+The nat table is:
+
+    {{index . "LNat4"}}
+
+<details>
+<summary>iptables commands</summary>
+
+    {{index . "SNat4"}}
+
+</details>
+
+Differences from [nat mode][1]:
+
+  - In the POSTROUTING chain:
+    - No MASQUERADE rule for traffic from the bridge network to elsewhere. [setupIPTablesInternal][2]
+    - No MASQUERADE rule for traffic from the bridge network to itself on published port 80 (port
+      mapping is skipped). [attemptBindHostPorts][3]
+  - In the DOCKER chain:
+    - No early return ("skip DNAT") for traffic from the bridge network. [setupIPTablesInternal][4]
+    - No DNAT rule for the published port (port mapping is skipped). [attemptBindHostPorts][3]
+
+_And, the userland proxy won't be started for mapped ports._
```

Comment:
I wonder if we can generate this/include this from the test instead of just having it directly in the template. Benefit would be to have one less place to have to update when making changes.

Not a blocker for sure, just a thought for future improvements.

Reply:
Yes, there's definitely scope for putting more of the words into the code itself.

In particular, the links have a chicken-and-egg problem - no way to get a permalink to the code until it's pushed, so there's no way to link to changed code. In later PRs, I linked to code in my clone or removed links and just named functions ... when this series of iptables changes are merged I'll make a change to update links.

Once we've refactored, ready to add native nftables support, the code should be less scattered and more stable. Then this should all get a bit easier.